x/swaprouter/router_test.go

```diff
@@ -941,3 +942,84 @@ func (suite *KeeperTestSuite) calcInAmountAsSeparateSwaps(osmoFeeReduced bool, r
 		return nextTokenIn
 	}
 }
+
+func (suite *KeeperTestSuite) TestSingleSwapExactAmountIn() {
+	tests := []struct {
+		name                   string
+		poolId                 uint64
+		poolCoins              sdk.Coins
+		poolFee                sdk.Dec
+		tokenIn                sdk.Coin
+		tokenOutDenom          string
+		tokenOutMinAmount      sdk.Int
+		expectedTokenOutAmount sdk.Int
+		expectError            bool
+	}{
+		{
+			name:                   "One route: Swap - [foo -> bar], 1 percent fee",
```

Comment:
nit: there can only be one route
```suggestion
			name:                   "Swap - [foo -> bar], 1 percent fee",
```